data/src/test/java/org/apache/iceberg/TestPartitionStatsHandler.java

```diff
@@ -82,6 +65,11 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.api.io.TempDir;
 
+/**
+ * This test covers {@link PartitionStatsHandler} from the core module. Since it relies on {@link
+ * InternalData}, and we cannot introduce a Parquet dependency in the core module, the test is
+ * placed externally.
+ */
```

Comment:
Asked about this in original `InternalData` PR https://github.com/apache/iceberg/pull/12060#issuecomment-2850881632